### ISSUE TITLE
Omit plugin-proposal-dynamic-import when using preset-env running @babel/cli

### ISSUE THREAD

**Issue by AdamRamberg** [babel/babel#10194]
## Feature Request

**Is your feature request related to a problem? Please describe.**
The problem I'm having started after #10109 was released. 

I want to use preset-env to transform static ES modules, but not dynamic imports. Dynamic imports should be left untouched. My case is that I'm building shared components / packages that should be transformed before distributed. However, in the consuming applications using the shared packages above I want Webpack to use the dynamic imports for code splitting. I guess that I could omit transforming our shared packages and then transform them in the consuming packages, however I would like to be able to support certain browsers / environments "out of the box" in the the shared packages. 

**Describe the solution you'd like**
Make it possible to send in `caller `options to the CLI (supportsDynamicImport and supportsStaticESM). Currently the `caller` option is getting overridden by the CLI, see [this](https://github.com/babel/babel/blob/master/packages/babel-cli/src/babel/util.js). Caller `name` should still always be overriden by the CLI. Can't think of any drawbacks supporting this behavior. 

**Describe alternatives you've considered**
Another solution would be to support excluding `plugin-proposal-dynamic-import` via the `preset-env` config. This however seems a little redundant, since we already have `supportsDynamicImport`.

**Teachability, Documentation, Adoption, Migration Strategy**
**Usage from command line**
`babel --caller '{supportsDynamicImport: false}'`
**Usage in code**
```
const parseArgv = require('@babel/cli/lib/babel/options').default;
const dirCommand = require('@babel/cli/lib/babel/dir').default;
const fileCommand = require('@babel/cli/lib/babel/file').default;

const { babelOptions, cliOptions } = parseArgv(process.argv);

babelOptions.caller = { supportsDynamicImport: false }; // This will now be taken into affect
const babelEntry = cliOptions.outDir ? dirCommand : fileCommand;
const babelResult = await babelEntry({ babelOptions, cliOptions });
```

**Comment by nicolo-ribaudo** [comment]
Doesn't `prset-env`'s `exclude` option work for your use case?

**Comment by AdamRamberg** [comment]
@nicolo-ribaudo Nope doesn't work. First of all it throws when it [normalizes the options](https://github.com/babel/babel/blob/master/packages/babel-preset-env/src/normalize-options.js) because it checks against [this list of plugins](https://github.com/babel/babel/blob/master/packages/babel-preset-env/data/plugins.json), which doesn't contain `plugin-proposal-dynamic-import`. Secondly, if it was included in the list of plugins it wouldn't work anyways. Both `plugin-proposal-dynamic-import` and `plugin-syntax-dynamic-import` are getting special treatment [here](https://github.com/babel/babel/blob/master/packages/babel-preset-env/src/index.js), and are thereby not affected by the exclude option. 

Now when I come back to it, it might be a better solution to just make it possible to exclude `plugin-proposal-dynamic-import` and `plugin-syntax-dynamic-import` 🤔 @nicolo-ribaudo Thoughts?

**Comment by nicolo-ribaudo** [comment]
Yes, I'd consider it a bug that it isn't possible to exclude `plugin-proposal-dynamic-import`.
Excluding the syntax plugin doesn't make sense instead, since it is what allows Babel to _parse_ it.

**Comment by AdamRamberg** [comment]
Alright! Makes sense that excluding the syntax plugin isn't desirable. 

Haven't been contributing to Babel before, but would like to take a stab at it and create a PR for making it possible to exclude `plugin-proposal-dynamic-import`. Will take a look at it in the next coming days and submit a PR (if someone else is not beating me to it).

**Comment by nicolo-ribaudo** [comment]
Thanks!

**Comment by SebastienGllmt** [comment]
I run into the exact same problem and also tried to exclude and it didn't work 😢

**Comment by nicolo-ribaudo** [comment]
As a workaround, use `modules: false` and manually include the plugin to transform es modules.

**Comment by vivek12345** [comment]
@nicolo-ribaudo is this something that I can pick up?

**Comment by AdamRamberg** [comment]
@vivek12345 the issue has been resolved in #10218, which has been approved by @JLHwung, but has not been merged to master yet. Not sure if something else is needed to get it merged though.

**Comment by VivekNayyar** [comment]
Alright. I actually missed the PR part. Didn't notice it has a PR already. Thanks for pointing that out.
